File: app/src/main/java/com/settlex/android/presentation/onboarding/OnboardingPage4Fragment.java
package com.settlex.android.presentation.onboarding;

import android.os.Bundle;
import android.text.SpannableString;
import android.text.Spanned;
import android.text.style.ForegroundColorSpan;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;

import com.bumptech.glide.Glide;
import com.settlex.android.R;
import com.settlex.android.databinding.FragmentOnboardingPage4Binding;

public class OnboardingPage4Fragment extends Fragment {
    private FragmentOnboardingPage4Binding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentOnboardingPage4Binding.inflate(inflater, container, false);

        highlightWordInHeader();
        setImage();

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    private void highlightWordInHeader() {
        String fullText = "Secure By\nDesign";
        String textToHighlight = "Design";
        int startIndex = fullText.indexOf(textToHighlight);
        int endIndex = startIndex + textToHighlight.length();

        SpannableString header = new SpannableString(fullText);
        header.setSpan(
                new ForegroundColorSpan(ContextCompat.getColor(requireContext(), R.color.blue_500)),
                startIndex,
                endIndex,
                Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
        );
        binding.header.setText(header);
    }

    private void setImage() {
        Glide.with(requireContext())
                .load(R.drawable.img_intro_slide_4)
                .centerCrop()
                .into(binding.ivIntroSlide4);
    }
}

File: app/src/main/java/com/settlex/android/presentation/onboarding/OnboardingPage4Fragment.kt
package com.settlex.android.presentation.onboarding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.settlex.android.R
import com.settlex.android.databinding.FragmentOnboardingPage4Binding
import com.settlex.android.presentation.common.util.SpannableTextFormatter

class OnboardingPage4Fragment : Fragment() {
    private var _binding: FragmentOnboardingPage4Binding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentOnboardingPage4Binding.inflate(inflater, container, false)

        highlightWordInHeader()
        setImage()

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    private fun highlightWordInHeader() {
        binding.header.text = SpannableTextFormatter(
            requireContext(),
            "Secure By\nDesign",
            "Design",
            R.color.colorPrimary,
        )
    }

    private fun setImage() {
        Glide.with(requireContext())
            .load(R.drawable.img_intro_slide_4)
            .centerCrop()
            .into(binding.ivIntroSlide4)
    }
}